t/custom-formats.t

```diff
@@ -0,0 +1,49 @@
+use lib '.';
+use t::Helper;
+use Mojo::Util 'decode';
+use Test::More;
+
+my $palindrome_schema = {
+  type       => 'object',
+  properties => {v => {type => 'string', format => 'palindrome'}}
+};
+
+{
+  my $check_palindrome = sub {
+    return $_[0] eq reverse($_[0]) ? undef : 'Not a palindrome.';
+  };
+
+  validate_custom_format_ok {v => 'Able was I ere I saw elbA'},
+    $palindrome_schema, 'palindrome', $check_palindrome;
+
+  validate_custom_format_ok {v => 'Amoeba'}, $palindrome_schema, 'palindrome',
+    $check_palindrome, E('/v', 'Not a palindrome.');
+}
```

Comment:
You can write this test in this way instead:

```
jv->formats->{palindrome} = sub {
  return $_[0] eq reverse($_[0]) ? undef : 'Not a palindrome.';
};

my $palindrome_schema = {
  type       => 'object',
  properties => {v => {type => 'string', format => 'palindrome'}}
};

validate_ok {v => 'Able was I ere I saw elbA'}, $palindrome_schema;
validate_ok {v => 'Amoeba'}, $palindrome_schema, $check_palindrome,
  E('/v', 'Not a palindrome.');
```

You can skip the block around the test, since none of the tests are overlapping.